code/go/0chain.net/smartcontract/storagesc/stakepool.go

```diff
@@ -436,6 +439,20 @@ func (ssc *StorageSmartContract) stakePoolUnlock(
 			"can't get related stake pool: %v", err)
 	}
 
+	dp, ok := sp.Pools[spr.PoolID]
+	if !ok {
+		return "", common.NewErrorf("stake_pool_unlock_failed", "no such delegate pool: %v ", spr.PoolID)
+	}
+
+	// if StakeAt has valid value and lock period is less than MinLockPeriod
+	if dp.StakedAt > 0 {
+		stakedAt := common.ToTime(dp.StakedAt)
+		minLockPeriod := config.SmartContractConfig.GetDuration("stakepool.min_lock_period")
+		if !stakedAt.Add(minLockPeriod).Before(time.Now()) {
```

Comment:
Use `After` could be more intuitive, so could be `if stakedAt.Add(minLockPeriod).After(time.Now()) {`

Reply:
@peterlimg  `!Before` means `Equals` or `After`